### PR TITLE
csfilter-kfp: script to filter known false positives

### PR DESCRIPTION
**Pull request by kdudka**
Resolves: https://issues.redhat.com/browse/OSH-736

### CODE REVIEW

File: src/csfilter-kfp [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+def construct_git_cmd(kfp_git_url):
+    # split kfp_git_url into the clone URL and (optional) revision
+    m = re.match("^(.*)#([0-9a-f]+)", kfp_git_url)
+    if m:
+        # checkout a specific revision
+        url = shlex.quote(m.group(1))
+        rev = m.group(2)
+        return f"git clone {url} ${{td}}/kfp\n" \
+               f"(cd ${{td}}/kfp && git reset -q --hard {rev})\n"
```
Comment (lzaoral) [inline src/csfilter-kfp:62]: There is no need to create a subshell: `f'git -C "${{td}}/kfp" reset -q --hard {rev}'` (I've also added quoting for consistency with the rest of the code.)

File: src/csfilter-kfp [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    else:
+        # shallow clone of the default branch
+        url = shlex.quote(kfp_git_url)
+        return f"git clone --depth 1 {url} ${{td}}/kfp\n"
```
Comment (lzaoral) [inline src/csfilter-kfp:66]: Quote, `${td}` for consistency.

File: src/csfilter-kfp [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    parser.add_argument(
+        "--project-nvr",
+        help="Name-Version-Release (NVR) of the scanned project, used to find path exclusions")
+
+    parser.add_argument(
+        "--record-excluded",
+        help="file to store all excluded findings to")
+
+    parser.add_argument(
+        "--json-output", action="store_true", default=(not os.isatty(sys.stdout.fileno())),
```
Comment (lzaoral) [inline src/csfilter-kfp:201]: `not sys.stdout.isatty()`: https://docs.python.org/3/library/io.html#io.IOBase.isatty

File: src/csfilter-kfp [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+# along with csdiff.  If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+
+
+# if neither --kfp-dir nor --kfp-git-url is specified, use the known-false-positives RPM pacakge
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/csfilter-kfp]: Fixed in 8cf2326eb90e66d9265de1df53716d5bf30a14de.

File: src/csfilter-kfp (hunk 1 above)
Comment (kdudka) [reply src/csfilter-kfp]: Fixed in f44a927f5dec86f843d57b2ece6b1005c8755d57.

File: src/csfilter-kfp (hunk 2 above)
Comment (kdudka) [reply src/csfilter-kfp]: Fixed in f44a927f5dec86f843d57b2ece6b1005c8755d57.

File: src/csfilter-kfp (hunk 3 above)
Comment (kdudka) [reply src/csfilter-kfp]: Fixed in 643b22983d2d47004fc6a938b84a87e4f6318cb1.

**Review by lzaoral — APPROVED** [review]

**Comment by kdudka** [comment]
@lzaoral Thank you for the review!

**Pull request merged**